src/sentry/release_health/metrics.py

```diff
@@ -606,3 +601,342 @@ def extract_row_info(row: Mapping[str, Union[OrganizationId, str]]) -> ReleaseNa
             return reverse_tag_value(organization_id, row.get(release_column_name))  # type: ignore
 
         return {extract_row_info(row) for row in result["data"]}
+
+    @staticmethod
+    def _get_session_duration_data_for_overview(
+        where: List[Union[BooleanCondition, Condition]], org_id: int
+    ) -> Mapping[Tuple[int, str], Any]:
+        """
+        Percentiles of session duration
+        """
+        rv_durations: Mapping[Tuple[int, str], Any] = {}
+
+        release_column_name = tag_key(org_id, "release")
+        aggregates: List[SelectableExpression] = [
+            Column(release_column_name),
+            Column("project_id"),
+        ]
+
+        for row in raw_snql_query(
+            Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity("metrics_distributions"),
+                select=aggregates + [Column("percentiles")],
+                where=where
+                + [
+                    Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session.duration")),
+                ],
+                groupby=aggregates,
+            ),
+            referrer="release_health.metrics.get_session_duration_data_for_overview",
+        )["data"]:
+            # See https://github.com/getsentry/snuba/blob/8680523617e06979427bfa18c6b4b4e8bf86130f/snuba/datasets/entities/metrics.py#L184 for quantiles
+            key = row["project_id"], reverse_tag_value(org_id, row[release_column_name])
+            rv_durations[key] = {
+                "duration_p50": row["percentiles"][0],
```

Comment:
Couldn't this raise an `IndexError`?

Reply:
snuba apparently returns a fixed-size array here. if that changes we should raise an indexerror